portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,10 +34,55 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private String name;
+  ArrayList<String> comments = new ArrayList<String>();
+  List<Task> tasks = new ArrayList<>();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = "Bless";
+    String json = convertToJson();
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Task task = new Task(id, title, timestamp);
+      tasks.add(task);
+    }
+  }
+  private String convertToJson() {
+    Gson gson = new Gson();
+    String json = gson.toJson(tasks);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println(name);
+    response.sendRedirect("/?authuser=0#comments");
```

Comment:
do we need the `/?authuser=0` part? I am not sure this works with the page jump, as the trailing `/` essentially assume it's a new page.

Try `#comments` instead?

Reply:
So when I removed it, it takes me to this page. 
![Screenshot 2020-06-23 at 5 05 14 PM](https://user-images.githubusercontent.com/44912212/85464772-06054b00-b576-11ea-8666-d7a91980ef97.png)

When I add the /?authuser=0#comments back it refreshes to the comments section

Reply:
Do you still get the same error when you deploy?

Reply:
It refreshes for a few seconds and returns to the same comments section but with no added comments when deployed with "#comments".

Reply:
That's because you're setting the response on the Servlet side before adding / querying the comments from the DB. I added a comment in DataServlet on how to fix this issue